cli/schema_list.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/spf13/cobra"
+
+	httpapi "github.com/sourcenetwork/defradb/api/http"
+	"github.com/sourcenetwork/defradb/config"
+	"github.com/sourcenetwork/defradb/errors"
+)
+
+type schemaListResponse struct {
+	Data struct {
+		Result      string `json:"result"`
+		Collections []struct {
+			Name   string `json:"name"`
+			ID     string `json:"id"`
+			Fields []struct {
+				ID   string `json:"id"`
+				Name string `json:"name"`
+				Kind string `json:"kind"`
+			} `json:"fields"`
+		} `json:"collections"`
+	} `json:"data"`
+	Errors []struct {
+		Message string `json:"message"`
+	} `json:"errors"`
+}
+
+func MakeSchemaListCommand(cfg *config.Config) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "list",
+		Short: "List schema types from DefraDB",
```

Comment:
suggestion: `List schema types with their respective fields`. No a strong opinion but I though it would be a bit more descriptive of what it does.